### PR TITLE
hacks to install `grpcio`

### LINKED ISSUE #10: Unable to install dbt@0.19.0rc2 on Big Sur (Version 11.1)
### Description

dbt fails to install with Homebrew. This occurs when Homebrew attempts to install `grpcio` with the following command:
```
==> /usr/local/Cellar/dbt@0.19.0-rc2/0.19.0rc2_1/libexec/bin/pip install -v --no-deps --no-binary :all: --no-user --ignore-installed /private/tmp/dbt-0.19.0-rc2--grpcio-20210117-98475-wurhff/grpcio-1.34.1
```

It produces the following error:
```
    Running command python setup.py egg_info
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/tmp/pip-req-build-gmlu06dy/setup.py", line 359, in <module>
        if mac_target and (pkg_resources.parse_version(mac_target) <
      File "/usr/local/Cellar/dbt@0.19.0-rc2/0.19.0rc2_1/libexec/lib/python3.8/site-packages/pkg_resources/__init__.py", line 128, in parse_version
        return packaging.version.Version(v)
      File "/usr/local/Cellar/dbt@0.19.0-rc2/0.19.0rc2_1/libexec/lib/python3.8/site-packages/pkg_resources/_vendor/packaging/version.py", line 219, in __init__
        match = self._regex.search(version)
    TypeError: expected string or bytes-like object
```

<details>
<summary>Config info</summary>

```
==> Formula
Tap: fishtown-analytics/dbt
Path: /usr/local/Homebrew/Library/Taps/fishtown-analytics/homebrew-dbt/Formula/dbt@0.19.0-rc2.rb
==> Configuration
HOMEBREW_VERSION: 2.7.5-10-g8869208
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8869208a601b7846f2e6597a49b4740cb3c06de2
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a9057054ff050df7d145ed2305dd68a0c1f2c7d6
Core tap last commit: 65 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 12.0 build 1200
Git: 2.30.0 => /usr/local/bin/git
/usr/bin/curl --version
Curl: 7.64.1 => /usr/bin/curl
/usr/libexec/java_home --xml --failfast
/usr/libexec/java_home --xml --failfast
Java: 15.0.1
macOS: 11.1-x86_64
CLT: 12.3.0.0.1.1607026830
Xcode: 12.3
```
</details>

### PR DESCRIPTION
**Pull request by kwigley**
Resolves #10 

There is an issue with the latest version of `grpcio` on macOS Big Sur, this should resolve the installation issue.

Tested locally on my machine
```console
$ sw_vers
ProductName:	macOS
ProductVersion:	11.1
BuildVersion:	20C69
```

### CODE REVIEW

File: Formula/dbt@0.19.0-rc2.rb [hunk 1]
```diff
@@ -391,14 +391,23 @@ def install
...
+        # adds the `--no-use-pep517` parameter
         r.stage {
           venv.instance_variable_get(:@formula).system venv.instance_variable_get(:@venv_root)/"bin/pip", "install",
             "-v", "--no-deps", "--no-binary", ":all:",
             "--ignore-installed", "--no-use-pep517", Pathname.pwd
         }
+      elsif r.name == "grpcio"
+        # workaround for installing `grpcio`, a dependency of `google-cloud-bigquery`
+        r.stage {
+          inreplace Pathname.pwd/"setup.py", "if mac_target and (pkg_resources.parse_version(mac_target) <", "if mac_target and (pkg_resources.parse_version(str(mac_target)) <" # https://github.com/grpc/grpc/pull/24998
```
Comment (kwigley) [inline Formula/dbt@0.19.0-rc2.rb:406]: This is the magical line, it replaces the buggy line in `setup.py` for `grpcio`

**Comment by jtcohen6** [comment]
The formula worked for me locally, I was able to successfully run `brew install Formula/dbt@0.19.0-rc2.rb`.

**Comment by kwigley** [comment]
In the spirit of being optimistic and not baking this one off fix into `dbt-release` scripts, I will just manually apply this patch to the new releases this issue affects. I like being hopeful and don't mind submitting a quick PR

**Comment by kwigley** [comment]
closing in favor of #12

**Pull request closed without merging**